plugins/coverbrowser.koplugin/listmenu.lua

```diff
@@ -388,16 +409,27 @@ function ListMenuItem:update()
 
             -- Build the right widget
 
-            local wfileinfo = TextWidget:new{
-                text = fileinfo_str,
-                face = Font:getFace("cfont", 14),
-                fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
-            }
-            local wpageinfo = TextWidget:new{
-                text = pages_str,
-                face = Font:getFace("cfont", 14),
-                fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
-            }
+            local fontsize_info = math.min(20, _fontSize(14))
+            local wfileinfo
+            local wpageinfo
+            -- Build file and page info texts with decreasing font size
+            -- till it fits in the space available
+            repeat
+                wfileinfo = TextWidget:new{
+                    text = fileinfo_str,
+                    face = Font:getFace("cfont", fontsize_info),
+                    fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
+                }
+                wpageinfo = TextWidget:new{
+                    text = pages_str,
+                    face = Font:getFace("cfont", fontsize_info),
+                    fgcolor = self.file_deleted and Blitbuffer.COLOR_DARK_GRAY or nil,
+                }
+                local height = wfileinfo:getSize().h + wpageinfo:getSize().h + Screen:scaleBySize(2)
+                fontsize_info = fontsize_info - 1
+            until height < dimen.h -- now it fits
```

Comment:
(Just a reminder: we still don't need that repeat...until.)

Reply:
and I'll cap the font size at something like Screen:height / 36.